src/Castle.Core/DynamicProxy/Generators/BaseProxyGenerator.cs

```diff
@@ -38,14 +38,19 @@ namespace Castle.DynamicProxy.Generators
 	internal abstract class BaseProxyGenerator
 	{
 		protected readonly Type targetType;
+		protected readonly Type[] interfaces;
 		private readonly ModuleScope scope;
 		private ILogger logger = NullLogger.Instance;
 		private ProxyGenerationOptions proxyGenerationOptions;
 
-		protected BaseProxyGenerator(ModuleScope scope, Type targetType, ProxyGenerationOptions proxyGenerationOptions)
+		protected BaseProxyGenerator(ModuleScope scope, Type targetType, Type[] interfaces, ProxyGenerationOptions proxyGenerationOptions)
 		{
+			CheckNotGenericTypeDefinition(targetType, "targetType");
+			CheckNotGenericTypeDefinitions(interfaces, "interfaces");
+
 			this.scope = scope;
 			this.targetType = targetType;
+			this.interfaces = TypeUtil.GetAllInterfaces(interfaces);
```

Comment:
Perhaps worth noting that one proxy type generator previously made this call *before*, not *after*, `CheckNotGenericTypeDefinitions`. I standardized the call order as seen here. Calling `TypeUtil.GetAllInterfaces` after the check should be more efficient since there are fewer interfaces to check; it should also be perfectly safe since a closed generic interface type cannot inherit an open generic interface type (AFAIK).